popup.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+
+let listeners = [];
+
+const tsSapCallFinishedMsg = 'call finished';
+
+chrome.runtime.sendMessage({
+	data: {
+		request: 'send logs'
+	}
+});
+
+chrome.runtime.onMessage.addListener(
+	function (request, sender, sendResponse) {
+		if (request.msg === tsSapCallFinishedMsg) {
+			const messagesElement = 'messages';
+			document.getElementById(messagesElement).innerHTML = formatServerResponse(request.data.serverResponse) +
+				document.getElementById(messagesElement).innerHTML;
+			const date = request.data.serverResponse.date;
+
+			retainOnlyNewestOfSubsequentLogEntries(messagesElement);
+			initializeResetLink(date);
+			initializeLogLink(date);
+			makeObjectInspectionCollapsible(date);
+			listeners.forEach(listenerEntry => {
+				document.getElementById(listenerEntry.elementId).addEventListener('click', listenerEntry.listener);
+			});
+		}
+	}
+);
+
+function formatServerResponse(serverResponse) {
+	let textParts = serverResponse.text.split('ManualSapTestInfo');
+	let text = textParts[0];
+	if (textParts.length > 1) {
+		text = textParts[0] +
+			'<div id="' + serverResponse.date + '-teaser">ManualSapTestInfo [<span style="color: darkgray">click to' +
+			' inspect</span>]</div>' +
+			'<div id="' + serverResponse.date + '-info" style="display: none">' +
+			'ManualSapTestInfo' + textParts[1].replace('[', '[<br>&nbsp;&nbsp;&nbsp;').replace(/, /g, '<br>&nbsp;&nbsp;&nbsp;') + '</div>';
+	}
+
+	text = text.replace(/(?:\r\n|\r|\n)/g, '<br>');
+
+	const resetUserStateUrl = 'api/projects/{project}/sap-test-event/reset/{sap-user}';
+	text = text.replace(resetUserStateUrl, resetUserStateUrl + ' (<span class="reset-link" id="' + serverResponse.date + '-reset">click' +
+		' here to send request now)</span>');
+
+	const testKeyPrefix = 'Test identifier: ';
+	const testKeyPostfix = ' get log via api/projects/{project}/sap-test-event/log/{sap-test-key}.';
+	text = text.replace(testKeyPrefix, '<span class="log-link" id="' + serverResponse.date + '-log" data-test-key="');
+	text = text.replace(testKeyPostfix, '">Click to retrieve/update log.</span>');
+
+	text.replace(':/CQSE/MSG_TIA:000 ', '');
+
+	let labelsClasses = 'labels';
+	if (serverResponse.status !== 200) {
+		labelsClasses += ' error';
+	}
+
+	return "<div class='" + labelsClasses + "'><div class='label'>" + serverResponse.action + "</div><div class='label'" +
```

Comment:
What I was thinking in my mind is something like this starting from line 43:

```
...
const formatValues = {
	'lineBreak': {
		'old': /(?:\r\n|\r|\n)/g,
		'new': '<br>'
	},
	'resetUserStateUrl': {
		'old': 'api/projects/{project}/sap-test-event/reset/{sap-user}',
		'new': resetUserStateUrl + ' (<span class="reset-link" id="' + serverResponse.date + '-reset">click' +
			' here to send request now)</span>'
	},
	'testKeyPrefix': {
		'old': 'Test identifier: ',
		'new': '<span class="log-link" id="' + serverResponse.date + '-log" data-test-key="'
	},
	'testKeyPostfix': {
		'old': ' get log via api/projects/{project}/sap-test-event/log/{sap-test-key}.',
		'new': '">Click to retrieve/update log.</span>'
	},
	'link': {
		'old': ':/CQSE/MSG_TIA:000 ',
		'new': ''
	}
}

text = text.replace(formatValues['lineBreak']['old'], formatValues['lineBreak']['new']);
text = text.replace(formatValues['resetUserStateUrl']['old'], formatValues['resetUserStateUrl']['new']);
text = text.replace(formatValues['testKeyPrefix']['old'], formatValues['testKeyPrefix']['new']);
text = text.replace(formatValues['testKeyPostfix']['old'], formatValues['testKeyPostfix']['new']);
text = text.replace(formatValues['link']['old'], formatValues['link']['new']);

...
```
In the previous one, the HTML and JS codes seem a bit mingled with each other which might make the interpretation difficult. Here, it is a bit clear which values will be replaced by which value, I think. 
Again your call :) If you want to go with this approach, please check the values one more because I quickly did the refactoring. And feel free to rename the variable names.